app/src/main/java/com/ifmo/balda/adapter/BoardGridAdapter.kt

```diff
@@ -0,0 +1,130 @@
+package com.ifmo.balda.adapter
+
+import android.graphics.Color
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.ToggleButton
+import com.ifmo.balda.R
+
+
+class BoardGridAdapter(
+  private val layoutInflater: LayoutInflater,
+  private val letters: MutableList<String>,
+  private val n: Int
+) : BaseAdapter() {
+
+  private var currentWord = linkedSetOf<Pair<ToggleButton, Int>>()
+
+  override fun getCount() = letters.size
+
+  override fun getItem(position: Int): Any? = null
+
+  override fun getItemId(position: Int): Long = 0
+
+  override fun getView(position: Int, grid: View?, parent: ViewGroup?): View {
+    var view = grid
+    if (view == null) {
+      view = (layoutInflater.inflate(R.layout.letter, null) as ToggleButton).apply {
+        setLetterValue()
+        isActivated = true
+        setOnClickListener(getOnLetterClickListener(position))
+        setOnTouchListener(getOnLetterTouchListener(this, position))
+      }
+    }
+    return view
+  }
+
+  private fun ToggleButton.setLetterValue() {
+    text = "A"
+    textOn = "A"
+    textOff = "A"
+  }
```

Comment:
предлагаю сразу заюзать getItem, ну просто а почему нет, всё равно это надо будет в будущем сделать

```suggestion
  override fun getItem(position: Int): String = letters[position]

  override fun getItemId(position: Int): Long = position.toLong()

  override fun getView(position: Int, grid: View?, parent: ViewGroup?): View {
    var view = grid
    if (view == null) {
      view = (layoutInflater.inflate(R.layout.letter, null) as ToggleButton).apply {
        setLetterValue(position)
        isActivated = true
        setOnClickListener(getOnLetterClickListener(position))
        setOnTouchListener(getOnLetterTouchListener(this, position))
      }
    }
    return view
  }

  private fun ToggleButton.setLetterValue(position: Int) {
    val letter = getItem(position)
    text = letter
    textOn = letter
    textOff = letter
  }
```